src/System.Slices/System/Buffers/OwnedMemory.cs

```diff
@@ -177,26 +185,32 @@ Span<T> IMemory<T>.GetSpan(long id)
         #endregion
     }
 
-    interface IKnown
+    public interface IReferenceCounted
     {
         void AddReference(long id);
         void Release(long id);
     }
+    public interface IReadOnlyMemory<T> : IReferenceCounted
+    {
+        ReadOnlySpan<T> GetSpan(long id);
+
+        unsafe bool TryGetPointer(long id, out void* pointer);
 
-    internal interface IMemory<T> : IKnown
+        DisposableReservation Reserve(ref ReadOnlyMemory<T> memory);
+    }
+    public interface IMemory<T> : IReadOnlyMemory<T>
```

Comment:
yay :) Nice readonly inheritance 👍 List not being derived from ReadonlyList has always bugged me...